app/ui/src/app/integration/integration-support.service.ts

```diff
@@ -1,93 +1,67 @@
-import { Injector, Injectable } from '@angular/core';
-import { Http, Response, ResponseContentType, URLSearchParams } from '@angular/http';
-import { Restangular } from 'ngx-restangular';
+import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
-import { ErrorObservable } from 'rxjs/observable/ErrorObservable';
 
-import { RESTANGULAR_MAPPER } from '@syndesis/ui/app.module';
-import { ConfigService } from '@syndesis/ui/config.service';
+import { ApiHttpService } from '@syndesis/ui/platform';
 import { Action, Connection } from '@syndesis/ui/model';
 import { Integration } from './integration.model';
-import { log, getCategory } from '@syndesis/ui/logging';
+import { integrationSupportEndpoints } from './integration-support.api';
 
 @Injectable()
 export class IntegrationSupportService {
-  service: Restangular = undefined;
-  filterService: Restangular = undefined;
-  metadataService: Restangular = undefined;
-  mapperService: Restangular = undefined;
-  configService: ConfigService = undefined;
 
   constructor(
-    restangular: Restangular,
-    injector: Injector,
-    private http: Http,
-    private config: ConfigService
+    private apiHttpService: ApiHttpService,
   ) {
-    this.service = restangular.service('integration-support');
-    this.filterService = restangular.service('integrations');
-    this.metadataService = restangular.service('connections');
-    const restangularMapper = injector.get(RESTANGULAR_MAPPER);
-    this.mapperService = restangularMapper.service('java-inspections');
-    this.configService = config;
+    // nothing to do
   }
 
   getFilterOptions(dataShape: any): Observable<any> {
-    const url = this.filterService
-      .one('filters')
-      .one('options')
-      .getRestangularUrl();
-    return this.http.post(url, dataShape);
+    return this.apiHttpService.post(integrationSupportEndpoints.filterOptions, dataShape);
   }
 
-  requestPom(integration: Integration) {
-    const url = this.service
-      .one('generate')
-      .one('pom.xml')
-      .getRestangularUrl();
-    return this.http.post(url, integration);
+  getHistory(id: string): Observable<any> {
+    return this.apiHttpService.setEndpointUrl(integrationSupportEndpoints.history, { id: id }).get();
+  }
+
+  getDeployments(id: string): Observable<any> {
+    return this.apiHttpService.setEndpointUrl(integrationSupportEndpoints.deployments, { id: id }).get();
+  }
+
+  getDeployment(id: string, version: string): Observable<any> {
+    return this.apiHttpService.setEndpointUrl(integrationSupportEndpoints.deployment, { id: id, version: version }).get();
```

Comment:
Just a quick note, for future refactors like this: In object literal parameters whose properties and mapped variables do match in their naming, we can shorten things up a bit with this syntax:

`{ id: id, version: version }` can become `{ id, version }`

Reply:
I'll do that now too, thanks!